src/MiniCover.HitServices/HitService.cs

```diff
@@ -40,11 +41,16 @@ public void Exit()
             {
                 if (clearTestMethodCache)
                 {
-                    this.Save();
+                    AppDomain.CurrentDomain.ProcessExit += CurrentDomain_ProcessExit;
```

Comment:
Why do you have deadlock? if you do that we can't play on big project write would be too long

Reply:
Don't know, seems to be something specific from that project.
I will try a different approach instead of using ProcessExit

Reply:
you should look at how it's working to write in console from multi thread :)